File: lib/lib_timesys/utc2gps.f
*
*  procedure     UTC2GPS
*
      DOUBLE PRECISION FUNCTION UTC2GPS(UTC)
*********1*********2*********3*********4*********5*********6*********7**
*      
*                           Purpose
*
*  Return the GPS time in secons past J2000 given UTC time. The UTC time is
*  given in seconds past J2000.
*
*                           Input_Output Auguments
*
*  Name     Type    I/O     Descriptin
*  ----     ----    ---     --------------------------------------------
*  UTC      REAL*8  I       utc time given in seconds past J2000 
*  GPS      REAL*8  O       gps time given in seconds past J2000
*      
*  History:
*
*  Vesion 1.0
*    
*  Time         Author      Description
*  ----         ------      --------------------------------------------
*  07.06.01     S.J.Zhang   build this subroutine    
*
*********1*********2*********3*********4*********5*********6*********7**
c
c     Declaration_of_Varialbes
c
      IMPLICIT NONE
C
C     Declaration_of_Constants
C
      INTEGER       MAX_LEAP
      PARAMETER(    MAX_LEAP = 20 )
c
c     Declaration_of_Input_Output_Varialbles
c
      REAL*8        UTC, GPS
C
C     Declaration_of_External_Functions
C
      REAL*8        TAIUTC
c
c     return
c
      GPS = UTC + TAIUTC(UTC) - 19.0d0
c
c     FUNCTION VALUE
c
      UTC2GPS = GPS
c
      return
c
      end

File: tool/GraceRead/lib/utc2gps.c
#include "TimeLib.h"

  /* Purpose:  This function will return the tdt time corresponding to
     the UTC time */

real utc2gps(
		real utc,             /* Input: utc is in seconds past J2000 */
                table  leap)          /* Input: leap second table */
  /* Output: real (return val)   GPS in seconds */
{
  /* leap seconds table contains TAI-UTC */
   static char SccsId[] = "@(#) utc2gps.c       1.1 01/08/01";

   real r;
 
   r = (taiutc(utc, leap) + utc - 19.0);   /* This returns gps time. */
   return r;

} 


